planning/motion_velocity_smoother/src/motion_velocity_smoother_node.cpp

```diff
@@ -417,76 +410,82 @@ void MotionVelocitySmootherNode::onCurrentTrajectory(const Trajectory::ConstShar
   RCLCPP_DEBUG(get_logger(), "========================== run() end ==========================\n\n");
 }
 
+void MotionVelocitySmootherNode::updateDataForExternalVelocityLimit()
+{
+  if (prev_output_.empty()) {
+    return;
+  }
+
+  // calculate distance to insert external velocity limit
+  const double travel_dist = calcTravelDistance();
+  external_velocity_limit_dist_ -= travel_dist;
+  external_velocity_limit_dist_ = std::max(external_velocity_limit_dist_, 0.0);
+  RCLCPP_DEBUG(
+    get_logger(), "run: travel_dist = %f, external_velocity_limit_dist_ = %f", travel_dist,
+    external_velocity_limit_dist_);
+}
+
 TrajectoryPoints MotionVelocitySmootherNode::calcTrajectoryVelocity(
   const TrajectoryPoints & traj_input) const
 {
   TrajectoryPoints output{};  // velocity is optimized by qp solver
 
   // Extract trajectory around self-position with desired forward-backward length
-  const auto input_closest = motion_utils::findNearestIndex(
-    traj_input, current_pose_ptr_->pose, std::numeric_limits<double>::max(),
-    node_param_.delta_yaw_threshold);
+  const auto input_closest = findNearestIndexFromEgo(traj_input);
+
   if (!input_closest) {
     auto clock{rclcpp::Clock{RCL_ROS_TIME}};
     RCLCPP_WARN_THROTTLE(
-      get_logger(), clock, 5000, "Cannot find the closest point from input trajectory");
+      get_logger(), *clock_, 5000, "Cannot find the closest point from input trajectory");
```

Comment:
I suggest to change `*clock_` to `this->get_clock()`.

Reply:
see my answer.
https://github.com/autowarefoundation/autoware.universe/pull/1363#discussion_r930793157